### ISSUE TITLE
dead code elimination

### ISSUE THREAD

**Issue by Trass3r** [microsoft/llvm-mctoll#32]
Looks like a final DCE pass should be added?
Hello world becomes:
```ll
define dso_local i32 @main(i32 %arg1) {
entry:
  %0 = alloca i64, align 8
  %1 = bitcast [10 x i8]* @RO-String to i8*
  %2 = call i32 (i8*, ...) @printf(i8* %1, i32 %arg1)
  ret i32 0
}
```
Even the paper shows quite a few dead llvm instructions.

**Comment by Trass3r** [comment]
`zext 0` also seems to be common:
```ll
%1 = zext i32 0 to i64
store i64 %1, i64* %0
```

**Comment by bharadwajy** [comment]
Indeed the code generated by llvm-mctoll, at present, is not optimal. No effort is yet expended in making raised LLVM IR optimal. One train of thought for not yet running any optimization passes on the Module Functions is that the raised output is expected to be consumed and consequently optimized by the clang/llvm to generate native code (potentially for a non-x64 target). The thought is that any optimizations, such as DCE, would be part of this compilation process and is not (yet?) critical for the raised code. 

In situations where the raised IR is not intended to be compiled to a native target, I'd expect that one could use LLVM opt tool to run LLVM passes as desired to get a more optimized version.

**Comment by Trass3r** [comment]
True, that also occurred to me. But you gotta be careful, `opt -S -O3` easily optimizes away everything 😄
Even `opt -S -instcombine -dce` already removes quite some code. Could maybe be added to the readme.
Btw, maybe there could be a mctoll mode that outputs to stdout so it can be directly piped into opt.
